Testscripts/Windows/PERF-XDP-TCP-LATENCY.ps1

```diff
@@ -0,0 +1,198 @@
+            Add-Content -Value "$param" -Path $constantsFile
+        }
+        Write-LogInfo "constants.sh created successfully..."
+        Write-LogInfo (Get-Content -Path $constantsFile)
+        # Start XDP Installation
+        $installXDPCommand = @"
+bash ./XDPDumpSetup.sh 2>&1 > ~/xdpConsoleLogs.txt
+. utils.sh
+collect_VM_properties
+"@
+        Set-Content "$LogDir\StartXDPSetup.sh" $installXDPCommand
+        Copy-RemoteFiles -uploadTo $receiverVMData.PublicIP -port $receiverVMData.SSHPort `
+            -files "$constantsFile,$LogDir\StartXDPSetup.sh" `
+            -username $user -password $password -upload -runAsSudo
+        Copy-RemoteFiles -uploadTo $senderVMData.PublicIP -port $senderVMData.SSHPort `
+            -files "$constantsFile" `
+            -username $user -password $password -upload -runAsSudo
+        $testJob = Run-LinuxCmd -ip $receiverVMData.PublicIP -port $receiverVMData.SSHPort `
+            -username $user -password $password -command "bash ./StartXDPSetup.sh" `
+            -RunInBackground -runAsSudo
+        # Terminate process if ran more than 5 mins
+        # TODO: Check max installation time for other distros when added
+        $timer = 0
+        while ($testJob -and ((Get-Job -Id $testJob).State -eq "Running")) {
+            $currentStatus = Run-LinuxCmd -ip $receiverVMData.PublicIP -port $receiverVMData.SSHPort `
+                -username $user -password $password -command "tail -2 ~/xdpConsoleLogs.txt | head -1" -runAsSudo
+            Write-LogInfo "Current Test Status: $currentStatus"
+            Wait-Time -seconds 20
+            $timer += 1
+            if ($timer -gt 15) {
+                Throw "XDPSetup did not stop after 5 mins. Please check xdpConsoleLogs."
+            }
+        }
+
+        $currentState = Run-LinuxCmd -ip $receiverVMData.PublicIP -port $receiverVMData.SSHPort `
+            -username $user -password $password -command "cat ~/state.txt" -runAsSudo
+        if ($currentState -imatch "TestCompleted") {
+            # Start PERF test without XDP
+            $ResultDir = "$LogDir\WithoutXDP"
+            New-Item -Path $ResultDir -ItemType Directory -Force | Out-Null
+            Run_Lagscope_PERF $ResultDir
+
+            # Start XDPDump on client
+            $xdp_command = "cd ~/bpf-samples/xdpdump && ./xdpdump -i $iFaceName > ~/xdpdumpoutPERF.txt"
+            $testJobXDP = Run-LinuxCmd -ip $receiverVMData.PublicIP -port $receiverVMData.SSHPort -username $user `
+                -password $password -command $xdp_command -RunInBackground -runAsSudo
+            Write-LogInfo "XDP Dump process started with id: $testJobXDP"
+
+            # Start PERF test with XDP
+            $ResultDirXDP = "$LogDir\WithXDP"
+            New-Item -Path $ResultDirXDP -ItemType Directory -Force | Out-Null
+            Run_Lagscope_PERF $ResultDirXDP
+            # collect and compare result
+            $currentState = Compare_Result $ResultDir $ResultDirXDP
+        }

```

Comment:
This logic is not clearly separated from previous  '$currentState -imatch "TestCompleted"',  it's easy to be messed up in the future,  I suggest to add more else branch for a clear logic flow.  Because the 'Compare_Result'  only return 'TestCompleted' and 'TestFailed'.   I also suggest in your native powershell functions, using some self description expression in the function name, like 'IsLatencyBelowThreashold'

Reply:
I didn't get where things can be messed up in future. Also, about the name 'IsLatencyBelowThreshold' here we are comparing two lagscope output with a threshold value. So this function name will be a bit misleading.

Reply:
fixed